module/gui/CaptchaDialog.py

```diff
@@ -0,0 +1,257 @@
+        else:
+            self.imgDataBuffer.close()
+            errMsg = "setupCaptcha: Could not load the captcha image (1)"
+
+        if errMsg != None:
+            self.log.error(errMsg)
+            self.emit(SIGNAL("setMovie(QMovie *)"), QMovie())
+            self.infoLabel.setText("*** ERROR ***")
+            self.infoLabel.show()
+            self.lineEdit.hide()
+            self.submitBtn.hide()
+            self.ignoreBtn.hide()
+            self.closeBtn.show()
+
+    def setFree(self):
+        self.currentID = None
+        self.currentResultType = None
+        self.hide()
+        self.processing = False
+
+    def setupNoCaptcha(self):
+        pix = QApplication.style().standardIcon(QStyle.SP_MessageBoxInformation).pixmap(64, 64)
+        self.imgLabel.setPixmap(pix)
+        self.imgLabel.show()
+        self.infoLabel.setText("<b>" + _("There is no captcha waiting.") + "</b>")
+        self.infoLabel.show()
+        self.lineEdit.hide()
+        self.submitBtn.hide()
+        self.ignoreBtn.hide()
+        self.closeBtn.show()
+        self.imgLabel.unsetCursor()
+
+    def setupInteractiveCaptcha(self):
+        pix = QApplication.style().standardIcon(QStyle.SP_MessageBoxInformation).pixmap(64, 64)
+        self.imgLabel.setPixmap(pix)
+        self.imgLabel.show()
+        text =  "<b>"      + _("This captcha must be solved with the pyLoad server's webinterface!") + "</b>"
+        text += "<br><br>" + _("Interactive captchas (e.g. ReCaptcha / I'm not a robot) can't be solved")
+        text += "<br>"     + _("with the pyLoad Client. Only captchas which require text input")
+        text += "<br>"     + _("or a single click on a picture are supported.")
+        text += "<br><br>" + _("The default URL of the pyLoad server's webinterface is:")
+        text += "<br>"     + "http://127.0.0.1:8000"
+        self.infoLabel.setText(text)
+        self.infoLabel.show()
+        self.lineEdit.hide()
+        self.submitBtn.hide()
+        self.ignoreBtn.hide()
+        self.closeBtn.show()
+        self.imgLabel.unsetCursor()
+
+    def slotShow(self, nocaptcha):
+        if self.isFree():
+            if nocaptcha:
+                self.setupNoCaptcha()
+            else:
+                self.setupInteractiveCaptcha()
+        self.paintEventSignal = True
+        self.show()
+        self.update()
+
+    def slotShow_continue(self):
+        if self.geo is not None:
+            self.setGeometry(self.geo)
+        if self.adjSize:
+            self.adjustSize()
+        self.submitBtn.setFocus(Qt.OtherFocusReason)
+        self.raise_()
+        self.activateWindow()
+
+    def slotSubmitText(self):
+        text = unicode(self.lineEdit.text())
+        if not text: # empty string, ignored by api or core
+            return
+        tid = self.currentID
+        self.currentID = None
+        self.currentResultType = None
+        self.emit(SIGNAL("done"), tid, text)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['SIGNAL' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)